server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -491,21 +514,30 @@ public Consumer getHost(String guestId, Owner owner) {
      */
     @Transactional
     public VirtConsumerMap getGuestsHostMap(Owner owner, Set<String> guestIds) {
-        Disjunction guestIdCrit = Restrictions.disjunction();
-        for (String possibleId : Util.getPossibleUuids(guestIds.toArray(
-                new String [guestIds.size()]))) {
-            guestIdCrit.add(Restrictions.eq("guestId", possibleId).ignoreCase());
-        }
-        Criteria crit = currentSession()
-            .createCriteria(GuestId.class)
-            .createAlias("consumer", "gconsumer")
-            .createAlias("gconsumer.guestIdsCheckIns", "checkins")
-            .add(Restrictions.eq("gconsumer.owner", owner))
-            .addOrder(Order.desc("checkins.updated"))
-            .setProjection(Projections.property("consumer"));
+        List<Consumer> hypervisors = new ArrayList<Consumer>();
+        int fromIndex = 0;
+        int toIndex = fromIndex + MAX_IN_QUERY_LENGTH;
 
-        // Note: may contain duplicates but is sorted so they appear later:
-        List<Consumer> hypervisors = crit.add(guestIdCrit).list();
+        Object[] ids = Util.getPossibleUuids(guestIds.toArray(new String [guestIds.size()])).toArray();
```

Comment:
Can we ditch this entirely? In particular, the Util.getPossibleUuid bits that call Util.transformUuid.

iirc, that code was to handle some now old/unsupported version of vmware that generated bogus uuids. 

I believe it artificially inflates the number of guest uuids we search for. (double? worse?). 

I think even if we attempt this, I think some large portion of the uuid space may be eliminated by
being an invalid uuid. (ie, if they don't have a 4 for the first digit of the third partition ala https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_.28random.29)